test/constraint/encoding.test.ts

```diff
@@ -444,6 +444,201 @@ describe('constraints/encoding', () => {
     });
   });
 
+  describe('scalePropertiesSupportedByScaleType', () => {
+    describe('scaleBandSize', () => {
+      let encQ: EncodingQuery = {
+        channel: Channel.X,
+        field: 'A',
+        type: Type.QUANTITATIVE,
```

Comment:
In this test suite for `scalePropertiesSupportedByScaleType`, it seems like type is always `Q`.  Are you confident that the test are correct for non Q type too?  If not, test it for at least one of the scale properties?